src/cryptonote_core/service_node_list.cpp

```diff
@@ -69,14 +69,15 @@ namespace service_nodes
       return;
 
     uint64_t current_height = m_blockchain.get_current_blockchain_height();
-    LOG_PRINT_L0("Recalculating service nodes list, scanning blockchain from height " << m_state.height);
-    LOG_PRINT_L0("This may take some time...");
+    if (m_state.height == current_height)
+      return;
 
+    MGINFO("Recalculating service nodes list, scanning blockchain from height: " << m_state.height << " to: " << current_height);
     std::vector<std::pair<cryptonote::blobdata, cryptonote::block>> blocks;
     for (uint64_t i = 0; m_state.height < current_height; i++)
     {
       if (i > 0 && i % 10 == 0)
-          LOG_PRINT_L0("... scanning height " << m_state.height);
+          MGINFO("... scanning height " << m_state.height);
```

Comment:
This (and the Recalculataing message above) was a useful bit of log info to have at L0 for rescans, even with not running with any debugging, to let the user know a potentially long operation is actually doing something: particularly because while it happens everything else is locked out.  In theory we ought to hit this much less, but even just 10k blocks could take a minute.

Reply:
MGINFO always shows up at the default log level? My main concern here is that L0 prints as a warning. I added your suggestion that we store state 100 blocks back from the tip of the chain (in other PR) so keeping this as L0 would emit the message as a warning on startup every time which might concern operators.

Reply:
Oh, okay, I thought MGINFO was suppressed at default.  Sounds good then.